packages/contracts-bedrock/justfile

```diff
@@ -45,6 +45,22 @@ clean:
 test *ARGS: build-go-ffi
   forge test {{ARGS}}
 
+# Runs upgrade path variant of contract tests.
+# TODO: enforce that the op-deployer binary exists
+test-upgrade *ARGS: build-go-ffi
+  #!/bin/bash
+  if [ ! -f "../../op-deployer/bin/op-deployer" ]; then
+    echo "Error: op-deployer binary not found at ../../op-deployer/bin/op-deployer, you need to build it first"
```

Comment:
I also get an error when running `just build` in the op-deployer dir, which I have not tried debugging yet

```
$ just build     
go build -o bin/op-deployer cmd/op-deployer/main.go
# github.com/ethereum-optimism/optimism/op-chain-ops/state
../op-chain-ops/state/memory_db.go:17:20: cannot use (*MemoryStateDB)(nil) (value of type *MemoryStateDB) as vm.StateDB value in variable declaration: *MemoryStateDB does not implement vm.StateDB (wrong type for method AddBalance)
                have AddBalance("github.com/ethereum/go-ethereum/common".Address, *big.Int)
                want AddBalance("github.com/ethereum/go-ethereum/common".Address, *uint256.Int, tracing.BalanceChangeReason)
../op-chain-ops/state/memory_db.go:31:56: too many arguments in call to core.DeveloperGenesisBlock
        have (number, number, "github.com/ethereum/go-ethereum/common".Address)
        want (uint64, *"github.com/ethereum/go-ethereum/common".Address)
../op-chain-ops/state/encoding.go:333:19: address.Hash undefined (type *"github.com/ethereum/go-ethereum/common".Address has no field or method Hash)
../op-chain-ops/state/encoding.go:339:19: address.Hash undefined (type "github.com/ethereum/go-ethereum/common".Address has no field or method Hash)
../op-chain-ops/state/encoding.go:346:39: common.HexToAddress(address).Hash undefined (type "github.com/ethereum/go-ethereum/common".Address has no field or method Hash)
# github.com/ethereum-optimism/optimism/op-chain-ops/deployer
../op-chain-ops/deployer/deployer.go:73:18: undefined: backends.NewSimulatedBackendWithOpts
../op-chain-ops/deployer/deployer.go:74:12: undefined: backends.WithCacheConfig
../op-chain-ops/deployer/deployer.go:77:12: undefined: backends.WithGenesis
../op-chain-ops/deployer/deployer.go:86:12: undefined: backends.WithConsensus
error: Recipe `build` failed on line 2 with exit code 1
```

Reply:
that's an odd error. Looks like an issue with deps, maybe it's using the wrong go-ethereum version.

Reply:
That's what I thought too, but it's using the version specified by mise. It's likely related to https://github.com/ethereum-optimism/optimism/pull/13195#discussion_r1875987397 as I have not done much dev since the mise transition. If you are not getting this error let's assume it is related to https://github.com/ethereum-optimism/optimism/pull/13195#discussion_r1875987397 and resolve this thread